rfcs/0130-Implementing Checks API in tc-github while preserving Statuses API

```diff
@@ -0,0 +1,44 @@
+# RFC 130 - Implementing Checks API
+* Comments: [#130](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/130)
+* Proposed by: @djmitche
+
+# Summary
+
+We want to implement Checks API without dropping the support os Statuses API. Statuses API would be the default. Our users
+will have the freedom to choose which API they want by indicating their choice in `.taskcluster.yml`.
+
+## Motivation
+
+So GitHub has come up with a new API to create status indicators in its UI, called Checks.
+Checks API isn't in any way compatible with the current Statuses API - status created with one cannot be updated with another;
+they are two totally separate sets of data. Calling both APIs would result in two sets of status indicators in the UI.
+Any third party tools and integrations that rely on Statuses API to work would stop working after we switched to Checks API.
+
+We want to implement Checks because it seems to be the future, and because it gives a much richer output and a whole bunch 
+of controls for our users.
+
+At the same time, we want to preserve compatibility with Statuses because some of our users use tools and integration that 
+rely on them.
+
+# Details
+
+Currently we create a task and a status in a single function that listens
+to github webhooks. The idea is to create a task in separate function that listens to webhooks;
+then listen to the pulse message (`task defined`? `artifact created`?) on a certain queue that signals about creation of a task.
```

Comment:
almost certainly not artifact created.  We no longer send that for `s3` artifacts because it was incorrect, and docker-worker and generic-worker do not support uploading `blob` artifacts.  It also would only happen when trying to upload files, which would be more towards the end of a task.

There's [taskDefined](https://docs.taskcluster.net/docs/reference/platform/taskcluster-queue/references/events#taskDefined) which is probably the right event.